lib/amorail/entity.rb

```diff
@@ -85,5 +103,55 @@ def safe_create
     def reload_model(response)
     end
 
+    # this method removes nil values and empty arrays from params hash (deep)
+    def normalize_params(data)
+      return data unless data.is_a?(Hash)
+
+      compacted = {}
+      data.each do |key,val|
+        case val
+        when Numeric, String
+          compacted[key]=val
+        when Array
+          val.compact!
+          # handle custom keys
+          val=normalize_custom_fields(val) if key == :custom_fields
+          unless val.empty?
+            compacted[key] = val.map{|el| normalize_params(el) } 
```

Comment:
Space missing to the left of {.<br>Space between { and | missing.<br>Trailing whitespace detected.